chartfx-math/src/main/codegen/de/gsi/math/MathBaseGen.java

```diff
@@ -0,0 +1,431 @@
+    /**
+     * h_cgs * c_cgs [erg cm]
+     */
+    public static final double HC_CGS = H_CGS * C_CGS;
+    /**
+     * Molecular weight of dry air [kg kmol^-1 (or gm mol^-1)]
+     *  1976 US Standard Atmosphere,
+     *  also see http://atmos.nmsu.edu/jsdap/encyclopediawork.html
+     */
+    public static final double MW_AIR = 28.9644;
+    /**
+     *  [J kg^-1 K^-1]
+     */
+    public static final double RG_AIR = (1000.0 * R) / MW_AIR;
+    /**
+     *
+     */
+    public static final double R_UNCERTAINTY = R * ((K_UNCERTAINTY / K) + (N_A_UNCERTAINTY / N_A));
+
+    public static double log(double x) {
+        return java.lang.Math.log(x);
+    }
+
+    public static double log10(double x) {
+        return java.lang.Math.log10(x);
+    }
+
+    public static double aCosH(double arg0) {
+        if (arg0 == 0.0)
+            return 0.0;
+        double ax = java.lang.Math.abs(arg0);
+        return java.lang.Math.log(arg0 + ax * java.lang.Math.sqrt(1. - 1. / (ax * ax)));
+    }
+
+    public static double aSinH(double arg0) {
+        if (arg0 == 0.0)
+            return 0.0;
+        double ax = java.lang.Math.abs(arg0);
+        return java.lang.Math.log(arg0 + ax * java.lang.Math.sqrt(1. + 1. / (ax * ax)));
+    }
+
+    public static double aTanH(double arg0) {
+        return java.lang.Math.log((1 + arg0) / (1 - arg0)) / 2;
+    }
+
+    // codegen: double -> long
+    /**
+     * some integer math
+     *
+     * @param x input px
+     * @param y input py
+     * @return sqrt(px*px + py*py)
+     */
+    public static double hypot(double x, double y) {
+        return java.lang.Math.hypot(x, y);
+    }
+    //// end codegen
+
+    //// codegen: double -> float, int, long
+    public static double abs(double d) {
+        return (d >= 0) ? d : -d;
+    }
+    //// end codegen
+
+    public static short abs(short d) {
+        return (d >= 0) ? d : ((short) -d);
+    }
+
+    public static double aCos(double arg0) {
+        return java.lang.Math.acos(arg0);
+    }
+
+    public static double aSin(double arg0) {
+        return java.lang.Math.asin(arg0);
+    }
+
+    public static double aTan(double arg0) {
+        return java.lang.Math.atan(arg0);
+    }
+
+    public static double aTan2(double y, double x) {
+        if (x != 0)
+            return java.lang.Math.atan2(y, x);
+        else if (y == 0)
+            return 0;
+        else if (y > 0)
+            return PI / 2;
+        else
+            return -PI / 2;
+    }
+
+    public static double ceil(double x) {
+        return java.lang.Math.ceil(x);
+    }
+
+    public static int ceilNInt(double x) {
+        return nInt(java.lang.Math.ceil(x));
+    }
+
+    public static double cos(double arg0) {
+        return java.lang.Math.cos(arg0);
+    }
+
+    public static double cosH(double arg0) {
+        return java.lang.Math.cosh(arg0);
+    }
+
+    public static boolean even(long a) {
+        return !((a & 1) != 0);
```

Comment:
Codacy found an issue: [Use opposite operator instead of the logic complement operator.](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5780327)